### ISSUE TITLE
Accepting enhancement proposals as drafts

### ISSUE THREAD

**Issue by minrk** [jupyter/governance#13]
From @rgbkrk at https://github.com/jupyter/enhancement-proposals/issues/11:

> Rather than go back and forth in a PR to accept a proposal, how about we accept a PR and call it a draft state. When we've accepted it, we make a PR to update the status. If it's deprecated, not implemented, etc. we make another PR to address that.

@ellisonbg asked that the decision be handled over here.

We have a few open enhancement proposals that are in a "yes, this is a good idea, more details later" state, but haven't been merged because the criteria for that are unclear.

This would make it easier to merge earlier and iterate on ideas.

**Comment by jasongrout** [comment]
Yes, especially since the thing we are merging is a proposal, not a decision. I think it should pass the "is a good idea" test, and be relatively well thought out as an initial proposal.

**Comment by rgbkrk** [comment]
:+1: to my proposal to accepting proposals as proposals in draft form

:-1: to the thinking that we couldn't make the decision over on the enhancement proposals repo

**Comment by ellisonbg** [comment]
I think it is important to encode governance decisions in one place.
Otherwise it is very difficult to track these things down in a central
location.

On Thu, Feb 11, 2016 at 9:10 AM, Kyle Kelley <notifications@github.com>
wrote:

> [image: :+1:] to my proposal to accepting proposals as proposals in draft
> form
>
> [image: :-1:] to the thinking that we couldn't make the decision over on
> the enhancement proposals repo
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/governance/issues/13#issuecomment-182961775>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by jasongrout** [comment]
Yep, think of this as a precedence-setting decision, rather than just a one-time thing :).

**Comment by rgbkrk** [comment]
My thought on why we would keep it in EP is that for people viewing the project, they're keeping it in context with the repo they're looking at. We can `@` mention the steering council anywhere.

**Comment by minrk** [comment]
I don't think this is even necessarily a governance decision - it's just a process question about whether we can merge & iterate on incomplete proposals before accepting them or not. It doesn't change whether or how proposals are "accepted"

**Comment by fperez** [comment]
I like the zmq stages that [@rgbkrk described](https://github.com/jupyter/enhancement-proposals/issues/11#issuecomment-181046207).  

In this case, acceptance into `draft` status would mean that we're OK with the basic idea to the point of encouraging implementation.  But if that is the earliest status of acceptance, we should then convey during the proposal/discussion stage to the author a clear "sounds good, put together a draft implementation so we can at least see it and we'll then accept it in draft status", vs. "we really don't want to move forward with this idea at all, don't bother implementing".

So, the workflow would be:

- while in PR status, the EP is only being discussed, and the author(s) are potentially implementing something based on that discussion.
- acceptance moves it to the main repo, likely in draft mode (something could be small enough to be pushed straight to stable, though that sound unlikely to be the common case).

How does this sound?

**Comment by minrk** [comment]
@fperez :+1:

**Comment by rgbkrk** [comment]
@fperez :+1:

**Comment by damianavila** [comment]
@fperez :+1:
